include/h2o/balancer.h

```diff
@@ -32,7 +32,7 @@ extern "C" {
 
 typedef struct st_h2o_balancer_t h2o_balancer_t;
 
-typedef size_t (*h2o_balancer_selector)(h2o_balancer_t *balancer, h2o_socketpool_target_vector_t *targets, char *tried);
+typedef size_t (*h2o_balancer_selector)(h2o_balancer_t *balancer, h2o_socketpool_target_vector_t *targets, char *tried, void *socketpool_req_data);
```

Comment:
I would appreciate it if you could change the name of the last argument to something more opaque than `sockpool_req_data`, which implies that it is passing a "request data".

My understanding is that it is an argument to hash_get_key_cb and that it can be anything, depending on the implementation of the callback function.